### ISSUE TITLE
consume_less and LSTM dropout

### ISSUE THREAD

**Issue by mthmn20** [fchollet/keras#3820]
Hi

The [recurrent layer documentation](https://keras.io/layers/recurrent/) for the consume_less argument says that, when the value of consume_less is 'gpu', "RNN dropout must be shared for all gates, resulting in a slightly reduced regularization." Does this mean that only one of dropout_U or dropout_W is used. If so, which one?

Thanks

**Comment by carlthome** [comment]
Dropout is essentially flipping a coin for every weight. If heads, we do nothing, but it it's tails we _drop the connection_ by overriding the weight to zero for that batch update. Typically we'd flip a coin for every gate per RNN cell in a layer (one coin flip for the forget gate, one for the input gate, and so on), but with `consume_less='gpu'` you'll only get one coin flip for all the gates in a cell instead.

**Comment by mthmn20** [comment]
Got it, thanks!
